components/browser/domains/src/main/java/mozilla/components/browser/domains/DomainAutoCompleteProvider.kt

```diff
@@ -24,8 +25,32 @@ class DomainAutoCompleteProvider {
         const val CUSTOM_LIST = "custom"
     }
 
-    private var customDomains = emptyList<String>()
-    private var shippedDomains = emptyList<String>()
+    data class Domain(val protocol: String, val hasWww: Boolean, val host: String) {
```

Comment:
To enforce the usage of create() - would it make sense to make the constructor private or internal (if used in testing)?

Reply:
@pocmo I had that thought too, but I changed it at the last minute to have the functionality for the AutocompleteListFragment. There is probably (definitely) a cleaner way to do this.